### DISCUSSION TITLE
KSQL statement support for INSERT query

### DISCUSSION THREAD

**Discussion by MaharaJothi** [provectus/kafka-ui#1986] [Q&A] [answered]
Hi

I'm trying to make use of Kafka UI to run some KSQL queries on top of Kafka but unable to do so.

## Issue Description
I was able to run CREATE and SELECT queries successfully.  However, when I run INSERT query, it doesn't seem to work.  I don't get any response whether it got executed or not.  
When I check via Command line, it seems that it didn't get executed. 
[Query Statements](https://ksqldb.io/quickstart.html)

Is there any list of supported statements? or Can someone tell me how we can run INSERT query using Kafka UI?

Thanks in advance.

**Comment by Haarolean** [comment]
Hey, thanks for reaching out.
Can you please provide an example query which you did run?
Please note that you have to do them one by one, one statement per query.

**Reply by MaharaJothi** [reply]
Thanks for the response @Haarolean 
Below are the INSERT queries I've executed:

```
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('c2309eec', 37.7877, -122.4205);
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('18f4ea86', 37.3903, -122.0643);
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('4ab5cbad', 37.3952, -122.0813);
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('8b6eae59', 37.3944, -122.0813);
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('4a7c7b41', 37.4049, -122.0822);
INSERT INTO riderLocations (profileId, latitude, longitude) VALUES ('4ddad000', 37.7857, -122.4011);
```
Those are of the format specified [here](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/insert-values/)

I tried to execute them one by one like you mentioned but I didn't get any response in the UI.
I'm tried with Helm Chart: `kafka-ui:0.0.3` version and also `kafka-ui:v0.4.0`

**Reply by Haarolean** [reply]
@MaharaJothi can you check browser's developer's console please? Are there any failed requests? is there anything upon your statements execution?

**Reply by MaharaJothi** [reply]
@Haarolean  I don't get any error or info in the Developer's console for the successful execution of `CREATE queries`.
However, I do see `422 Unprocessable Entity` status for the POST request in Developer's console for `INSERT query`:
![image](https://user-images.githubusercontent.com/23254064/169212864-27544a6c-0165-44c7-a3e0-48984f456448.png)

## Response:
![image](https://user-images.githubusercontent.com/23254064/169212941-fbd4a6c7-bade-464d-baf9-3c3af3d1be51.png)

## Request:
![image](https://user-images.githubusercontent.com/23254064/169212973-560d636e-7786-43b6-b8fb-f0112e472a1e.png)

**Reply by MaharaJothi** [reply]
Please note, I am able to run the same set of queries manually via `KSQL CLI` successfully.  Its failing only from Kafka UI

**Reply by Haarolean** [reply]
@MaharaJothi please try newer version, yours is way outdated. Try either `latest` or `v0.4.0` tag.

**Reply by MaharaJothi** [reply]
@Haarolean I'm seeing this issue even with `v0.4.0`

**Reply by Haarolean** [reply]
@MaharaJothi any errors in logs? is there any error message for this requests' body in developer console? Also, have you tried chrome? We haven't tested this with FF.

**Reply by MaharaJothi** [reply]
@Haarolean  I tried in Chrome as well and seeing the same issue:
![image](https://user-images.githubusercontent.com/23254064/170638151-e8aabf81-9aba-41d5-8d97-d2324a63b95c.png)

I can see `200 OK` response for CREATE and SELECT queries.  But for `INSERT` I'm still seeing `422`
I don't see any error in the kafka-ui pod logs.  Can you please tell me how I can get logs for the specific query I executed?

**Reply by Haarolean** [reply]
@MaharaJothi you're still on the old version, ksql doesn't work properly in the version you're running. Please update to v0.4.0 and we can continue.

**Comment by Haarolean** [comment, marked as answer]
Provided a recommendation to [update](https://github.com/provectus/kafka-ui/discussions/1986#discussioncomment-2833292)
